subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -2436,8 +2436,54 @@ ssize_t ztls_sendmsg_ctx(struct tls_context *ctx, const struct msghdr *msg,
 	ssize_t ret;
 	int i;
 
+	if (msg == NULL) {
+		errno = EINVAL;
+		return -1;
+	}
+
 	if (IS_ENABLED(CONFIG_NET_SOCKETS_ENABLE_DTLS) &&
 	    ctx->type == SOCK_DGRAM) {
+		if (CONFIG_NET_SOCKETS_DTLS_SENDMSG_BUF_SIZE > 0) {
```

Comment:
I think this is an issue for building without `CONFIG_NET_SOCKETS_ENABLE_DTLS` as `CONFIG_NET_SOCKETS_DTLS_SENDMSG_BUF_SIZE` won't be defined.

Reply:
Good point, should be fixed now.